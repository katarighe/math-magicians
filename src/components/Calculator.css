.calculator {
  display: flex;
  flex-wrap: nowrap;
  max-width: 600px;
  padding: 0 95px;
  margin: 70px;
}

.calculator_header {
  align-items: center;
  justify-content: center;
}

.header {
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 3rem;
  animation: header-animation 2s infinite;
  color: orange;
}

@keyframes header-animation {
  0% {
    transform: translateY(0);
  }

  50% {
    transform: translateY(-10px);
  }

  100% {
    transform: translateY(0);
  }
}

.calculator_page {
  display: flex;
  justify-content: center;
  padding: 16px;
}

.calculator_container {
  width: 400%;
  margin: 0 auto;
}

.calculator_screen {
  padding: 15px;
  font-size: 3.75em;
  text-align: right;
  background-color: #808080;
  color: #fff;
}

.calculator_UI {
  display: flex;
  flex-wrap: wrap;
}

button {
  font-family: 'Segoe UI', sans-serif;
  flex: 0 1 25%;
  height: 70px;
  color: #111;
  font-size: 2em;
  background: #e0e0e0;
  border: 1px solid #808080;
  cursor: pointer;
}

.btn_opaque {
  background-color: #ffa500;
  color: #000;
}

.btn_large {
  flex: 0 1 50%;
}
