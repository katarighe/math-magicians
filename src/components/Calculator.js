import React, { Component } from 'react';

export default class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div>
        <div className="screen">
          <span>0</span>
        </div>
        <div className="buttons">
          <button type="button" value="clear">AC</button>
          <button type="button" value="pos-neg">+/-</button>
          <button type="button" value="%">%</button>
          <button type="button" value="/">รท</button>
          <button type="button" value="7">7</button>
          <button type="button" value="8">8</button>
          <button type="button" value="9">9</button>
          <button type="button" value="+">+</button>
        </div>
        <div className="buttons">
          <button type="button" value="4">4</button>
          <button type="button" value="5">5</button>
          <button type="button" value="6">6</button>
          <button type="button" value="-">-</button>
        </div>
        <div className="buttons">
          <button type="button" value="1">1</button>
          <button type="button" value="2">2</button>
          <button type="button" value="3">3</button>
          <button type="button" value="*">*</button>
        </div>
        <div className="buttons">
          <button type="button" value="0">0</button>
          <button type="button" value=".">.</button>
          <button type="button" value="C">C</button>
          <button type="button" value="/">/</button>
        </div>
      </div>
    );
  }
}
