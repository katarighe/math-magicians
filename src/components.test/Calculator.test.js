import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
