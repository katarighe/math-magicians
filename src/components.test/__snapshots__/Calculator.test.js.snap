// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if there is no items present 1`] = `
<div
  className="calculator_header"
>
  <h1
    className="header"
  >
    Let us do some math!
  </h1>
  <div
    className="calculator_page"
  >
    <div
      className="calculator"
    >
      <div
        className="calculator_container"
      >
        <div
          className="calculator_screen"
        >
          0
        </div>
        <div
          className="calculator_UI"
        >
          <button
            onClick={[Function]}
            type="button"
            value="clear"
          >
            AC
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="pos-neg"
          >
            +/-
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="%"
          >
            %
          </button>
          <button
            className="btn_opaque"
            onClick={[Function]}
            type="button"
            value="/"
          >
            รท
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="7"
          >
            7
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="8"
          >
            8
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="9"
          >
            9
          </button>
          <button
            className="btn_opaque"
            onClick={[Function]}
            type="button"
            value="*"
          >
            x
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="4"
          >
            4
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="5"
          >
            5
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="6"
          >
            6
          </button>
          <button
            className="btn_opaque"
            onClick={[Function]}
            type="button"
            value="-"
          >
            -
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="1"
          >
            1
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="2"
          >
            2
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="3"
          >
            3
          </button>
          <button
            className="btn_opaque"
            onClick={[Function]}
            type="button"
            value="+"
          >
            +
          </button>
          <button
            className="btn_large"
            onClick={[Function]}
            type="button"
            value="0"
          >
            0
          </button>
          <button
            onClick={[Function]}
            type="button"
            value="."
          >
            .
          </button>
          <button
            className="btn_opaque"
            onClick={[Function]}
            type="button"
            value="="
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
